### PR TITLE
Move to GitHub actions

### PR DESCRIPTION
**Pull request by MancunianSam**
There are three stages now.
* The WAF has the US added to it so that the GitHub actions runners can access the front end.
* Each feature file runs in it's own machine. Each run adds the machine IP to the security group and removes it after.
* The US is removed from the WAF.

This calls the new API to create Keycloak users so this PR is dependent on:
https://github.com/nationalarchives/tdr-keycloak-user-management/pull/1
https://github.com/nationalarchives/tdr-terraform-environments/pull/183
https://github.com/nationalarchives/tdr-terraform-modules/pull/169
https://github.com/nationalarchives/tdr-terraform-backend/pull/152

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment'
+        required: true
+        default: 'intg'
+env:
+  DRIVER_LOCATION: /usr/local/share/gecko_driver/geckodriver
```
Comment (TomJKing) [inline .github/workflows/ci.yml:11]: I think the current Jenkins job allows the selection of different browsers to run the test on?

**Review by TomJKing — COMMENTED** [review]
Looks good. One minor comment. Have you merged the latest master into this PR. Looks like a load of changes appearing that are already in the master branch?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MancunianSam) [reply .github/workflows/ci.yml]: Chrome doesn't work though :smile: You're right though, I've added in a python script to allow you to choose one. It's probably not the cleanest solution but it is working.

**Review by TomJKing — DISMISSED** [review]
Looks good.

File: src/test/resources/application.staging.conf [hunk 2]
```diff
@@ -3,7 +3,7 @@ include "application.base"
 tdr.base.url="https://tdr-staging.nationalarchives.gov.uk"
```
Comment (TomJKing) [inline src/test/resources/application.staging.conf:3]: Assuming the `keycloak.user.api.url` has not been set yet, as this still needs to be created in staging

**Review by TomJKing — COMMENTED** [review]
Looks good. Just one very minor question.

File: src/test/resources/application.staging.conf (hunk 2 above)
Comment (MancunianSam) [reply src/test/resources/application.staging.conf:4]: Ah good spot, I forgot about that, I've added it in.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**